### PR TITLE
Implement Json endpoint for fetching the number of free event tickets

### PR DESCRIPTION
**Pull request by afiore**
This PR demotes `memberTierFeatures` to a private method, and refactors it so that it accepts a subscription instead of a member object. Additionally, it updates `getUsageCountWithinTerm` changing its return type from plain `Int` to `Option[Int]`. In this case, the intended semantics for `None` is that the usage is not entitled to free event tickets, while `Some` implies instead that the user has such benefit.

@rtyley

### CODE REVIEW

**Comment by afiore** [comment]
@davidrapson I am waiting for @rtyley feedback on this, than I would hand this branch over to you for implementing the front-end part

File: frontend/conf/routes [hunk 1]
```diff
@@ -47,6 +47,7 @@ GET            /staff/event-details               controllers.Staff.eventDetails
 # Subscription
 POST           /subscription/update-card          controllers.Subscription.updateCard
 OPTIONS        /subscription/update-card          controllers.Subscription.updateCardPreflight
+GET            /subscription/tickets-left         controllers.Subscription.ticketsLeft
```
Comment (davidrapson) [inline frontend/conf/routes]: [URL design nerd warning]

This reads weirdly, maybe `ticket-allocation` or `remaining-tickets`

Comment (afiore) [inline frontend/conf/routes]: Warning accepted...

**Comment by davidrapson** [comment]
Added a few commits to actually show the usage on the front-end:

screen shot 2015-09-09 at 15 03 30

Not 100% ready as we would like to show a different message for first-time users.

**Comment by rtyley** [comment]
LGTM, though as discussed, we'll want to keep an eye on it, make sure it's not overloading Zuora :-)

**Comment by davidrapson** [comment]
Just added a couple of updates, so as long as they look OK we are ready to go:

- Updated logic to only show banner on Guardian Live events detail pages
- Updated messages so we show contextual messages depending on the state of the user
- Moved the banner markup into a fragment

**New user on listing page**

new-listing

**New user on event detail**
new-event

**Returning user on listing page**
returning-listing

**Returning user on event detail**
returning-event

@afiore @rtyley

**Comment by afiore** [comment]
I had a go at running it locally and it seems fine. Ready to merge unless anyone as something against that...

**Comment by davidrapson** [comment]
Go for it :shipit:

**Comment by afiore** [comment]
Verified on prod

screen shot 2015-09-10 at 16 03 22

**Comment by davidrapson** [comment]
:gem:

**Pull request merged**